### PR TITLE
Feature/gate changes

### PR DESCRIPTION
**Pull request by RyanM-RMA**
Adding gate change endpoints

### CODE REVIEW

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        try (HttpRequestResponse response = executor.execute()) {
+            return RadarObjectMapper.mapJsonToSetOfObjects(response.getBody(), GateChange.class);
+        }
+    }
+
+    public void storeGateChange(ApiConnectionInfo apiConnectionInfo, GateChangeEndpointInput.Post input)
+            throws IOException {
+        String endpoint = GATE_CHANGE_CREATE_PATH;
+        String body = RadarObjectMapper.mapObjectToJson(input.gateChanges());
+        new HttpRequestBuilderImpl(apiConnectionInfo, endpoint)
```
Comment (rma-bryson) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java:43]: need to set the accept-query-header `.addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1)`

**Review by rma-bryson — CHANGES_REQUESTED** [review]

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+
+    public static final class Post extends EndpointInput {
+        static final boolean DEFAULT_FAIL_IF_EXISTS = true;
+        static final String FAIL_IF_EXISTS = "fail-if-exists";
+        private boolean failIfExists = true;
+        private final Set<GateChange> changes = new HashSet<>();
+
+        public Post(Set<GateChange> changes) {
```
[the remark this answers is not in the document]
Comment (RyanM-RMA) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java]: Made private

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        private final String officeId;
+        private final String projectId;
+        private final Instant beginDate;
+        private final Instant endDate;
+        private boolean startInclusive = DEFAULT_START_INCLUSIVE;
+        private boolean endInclusive = DEFAULT_END_INCLUSIVE;
+        private String unitSystem = DEFAULT_UNIT_SYSTEM;
+        private int pageSize = DEFAULT_PAGE_SIZE;
+
+        public GetAll(String officeId, String projectId, Instant beginDate, Instant endDate) {
```
[the remark this answers is not in the document]
Comment (RyanM-RMA) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java]: Made private

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        static final String BEGIN = "begin";
+        static final String END = "end";
+        static final String OVERRIDE_PROTECTION = "override-protection";
+        private final String officeId;
+        private final String projectId;
+        private final Instant beginDate;
+        private final Instant endDate;
+        private boolean overrideProtection = DEFAULT_OVERRIDE_PROTECTION;
+
+        public Delete(String officeId, String projectId, Instant beginDate, Instant endDate) {
```
[the remark this answers is not in the document]
Comment (RyanM-RMA) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeEndpointInput.java]: Made private

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+public class GateChangeController {
+    static final String GATE_CHANGE_PATH = "projects/%s/%s/gate-changes";
+    static final String GATE_CHANGE_CREATE_PATH = "projects/gate-changes";
+
+    public Set<GateChange> retrieveGateChanges(ApiConnectionInfo apiConnectionInfo, GateChangeEndpointInput.GetAll input)
+            throws IOException {
+        String endpoint = format(GATE_CHANGE_PATH, input.officeId(), input.projectId());
+        HttpRequestExecutor executor = new HttpRequestBuilderImpl(apiConnectionInfo, endpoint)
+                .addEndpointInput(input)
```
[the remark this answers is not in the document]
Comment (RyanM-RMA) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java:33]: Added the accept query header

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java (hunk 1 above)
Comment (RyanM-RMA) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/GateChangeController.java]: Added the accept query header

**Review by rma-bryson — APPROVED** [review]

**Pull request merged**